chall03/alcohen.go

```
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var gTimeStart = time.Now()
+
+func elapsedTime() time.Duration {
+	return time.Since(gTimeStart) / 1e6
+}
+
+func makeGetRequest(url string) string {
+	resp, err := http.Get(url)
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	defer resp.Body.Close()
+	return string(body)
+}
+
+func generateHexRequestURL(url string, id string, respHex string) string {
+	return url + "?id=" + id + "&resp=" + respHex
+}
+
+func main() {
+	url := "https://chall03.hive.fi/"
+	fmt.Printf("%dms - GET %s\n", elapsedTime(), url)
+
+	body := makeGetRequest(url)
+	fmt.Printf("%dms - GET %s\n", elapsedTime(), url)
+	fmt.Printf("\tanswer:%s\n", body)
+
+	s := strings.Split(body, "=")
+	id := strings.Split(s[1], ",")[0]
+	r, _ := strconv.Atoi(strings.Split(s[2], ",")[0])
+	g, _ := strconv.Atoi(strings.Split(s[3], ",")[0])
+	b, _ := strconv.Atoi(strings.Split(s[4], " ")[0])
```

you should not ignore errors here.